Paws4Me/Paws4Me/Display All Adoptable Pets/View Models/PetViewModel.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  PetViewModel.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/24.
+//
+
+import Foundation
+
+// MARK: - PetViewModel Delegate
+protocol PetViewModelDelegate: AnyObject {
+    func reloadView()
+    func show(error: String)
+}
+
+class AllPetDataViewModel {
+
+    // MARK: - Vars/Lets
+    private var petRepository: PetDataRepository?
+    private weak var delegate: PetViewModelDelegate?
+    private var filteredPetObject: AdoptPet?
+    private var adoptPetObject: AdoptPet?
+    private var isSingleSearch = false
+    private var isFilterSearch = true
+    private var indexSinglePet: Int?
+    private var selectedGender: String = ""
+    private var animalType = ""
+
+    // MARK: - Constructor
+    init(repository: PetDataRepository,
+         delegate: PetViewModelDelegate) {
+        self.petRepository = repository
+        self.delegate = delegate
+    }
+
+    // MARK: - Functions
+    // MARK: - Functions Fetch Json Object
+    func fetchPetDataResults() {
+        petRepository?.fetchPetDataResults(method: .GET, endpoint: Constants.adoptURL) { [weak self] result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success(let petData):
+                    self?.adoptPetObject = petData
+                    self?.filteredPetObject = petData
+                    self?.delegate?.reloadView()
+                case .failure(let error):
+                    self?.delegate?.show(error: error.rawValue)
+                }
+            }
+        }
+    }
+
+    // MARK: - Tableview Functions
+    var petCount: Int {
+        return filteredPetObject?.page?.count ?? 0
+    }
+
+    var objectFilteredPet: AdoptPet? {
+        return filteredPetObject
+    }
+
+    var singleSearch: Bool {
+        return isSingleSearch
+    }
+
+    var filterSearch: Bool {
+        return isFilterSearch
+    }
+
+    var singlePetIndex: Int? {
+        return indexSinglePet
+    }
+
+    // MARK: - Search Pet by name
+    func setPetSearchName(petSearchText: String?) {
+        guard let searchPetText = petSearchText?.lowercased()  else {
+            filteredPetObject = adoptPetObject
+            return
+        }
+
+        searchPetName(searchText: searchPetText)
+    }
+
+    func searchPetName(searchText: String) {
+        isSingleSearch = true
+        guard let petObject = adoptPetObject else { return }
+
+        filteredPetObject?.page = petObject.page?.filter { $0.name?.lowercased().starts(with: searchText)
+            ??  false}
+
+        guard let filteredPet = filteredPetObject else { return }
+        setIndexForSpecificPetName(searchText: searchText, filteredPetObject: filteredPet)
+
+        if filteredPetObject == nil {
+            filteredPetObject = adoptPetObject
+        }
+    }
+
+    func setIndexForSpecificPetName (searchText: String, filteredPetObject: AdoptPet) {
```

Comment:
`func setIndexForSpecificPetName(searchText: String, filteredPetObject: AdoptPet)`

Spacing issue 😄

Reply:
Thank you 👍🏻